src/pages/Document.tsx

```diff
@@ -40,18 +106,21 @@ export default function Document() {
     return () => {
       removeStrokeEndListener(updateData);
     };
-  }, [doc, addStrokeEndListener, removeStrokeEndListener]);
+  }, [doc, addStrokeEndListener, removeStrokeEndListener, dataArray]);
```

Comment:
I see that you have used stroke end listener for handling doc update. It seems that we need to modify the data change listener for collab-dotting